src/idl_gen_json_schema.cpp

```diff
@@ -123,31 +123,44 @@ class JsonSchemaGenerator : public BaseGenerator {
     return path + file_name + ".schema.json";
   }
 
+  // If indentation is less than 0, that indicates we don't want any newlines
+  // either.
+  const std::string NewLine() {
```

Comment:
You also decided to refactor the code using these 2 functions. I personally find the code less readable this way, but I guess I don't particularly care either way.

Reply:
I tried to leave the code better than before. Indentation with a spaces was unreadable for me. So I came to this solution. This is also done in similar way in idl_gen_text.cpp.

Reply:
I appreciate the effort. Yes, generators written by different people with different ideas on what's readable :(